dask/array/reductions.py

```diff
@@ -607,6 +607,9 @@ def cumreduction(func, binop, ident, x, axis, dtype=None):
     if dtype is None:
         dtype = func(np.empty((0,), dtype=x.dtype)).dtype
     assert isinstance(axis, int)
+    if axis < 0:
+        axis = axis + x.ndim
+    assert axis >= 0
```

Comment:
This should be a ValueError, not an assert (this is external API, not an internal correctness check)